index.js

```diff
@@ -0,0 +1,108 @@
+const fs = require('fs')
+const fetch = require('node-fetch');
+const chalk = require('chalk')
+
+function allTheLinks() {
+    let newFile = process.argv.indexOf('--file');
+    let uri = process.argv[newFile + 1];
+    let probe = uri.lastIndexOf(".");
+    let typeFile = uri.slice(probe);
+    //console.log(typeFile);
+    if (typeFile === ".md") {
+        //console.log ("ahí vámos")
+        read(uri) //para leer el archivo
+            .then(content => findLinks(content) //para extraer los links
+                .then(linksReal => getStatusLink(linksReal))
+
+            ).catch(er => console.log(er))
+    }
+    else { console.log("you need a file with an .md extention ") }
+}
+allTheLinks();
+
+
+function read(file) {
+    return new Promise(function (resolve, reject) {
+        fs.readFile(file, function (err, content) {
+            if (err) {
+                return reject(err)
+            }
+            resolve(content)
+        })
+    })
+}
+
+const findLinks = (datos) => {
+    let joinArray = []
+    return new Promise((resolve) => {
+        let regularE = /(ftp|http|https):\/\/.*\)/gi;
+        let noStr = datos;
+        let str = noStr.toString();
+        let myArray;
+        while ((myArray = regularE.exec(str)) !== null) {
+
+            // let joinData= myArray[0].replace(")", " ")
+            let joinData = {
+                link: myArray[0].replace(")", " "),
+
+            };
+            joinArray.push(joinData);
+        }
```

Comment:
Dime que entiendes este while plis, y porqué no un for?

Reply:
estoy usando el  while,, para que mientras el arreglo que me regresa excec sea diferente de nulo (osea que si tenga una coincidencia con mi expresión regular), me vaya juntando las coincidencias para  hacerlas un sólo array, si podría usar un for pero tendría que replantearlo, le tendría que decir que lo que me regresa excec lo recorra y después lo junte